### PR TITLE
chore: bump automation

### PR DESCRIPTION
**Pull request by chesedo**
Having messed up the release today, I decided to just create scripts to make the future easier....

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    cargo set-version --workspace $version
+    git commit -am "chore: v$version"
+}
+
+function update-examples-versions()
+{
+    local version=$1
+
+    rg "shuttle-service = \{ version" --files-with-matches -g '!www/*' | xargs sed -i "s/shuttle-service = { version = \"[[:digit:]]*.[[:digit:]]*.[[:digit:]]*\"/shuttle-service = { version = \"$version\"/g"
```
Comment (brokad) [inline release.sh:22]: This strongly depends on the formatting of the `shuttle-service` entry in the `Cargo.toml` - would it be potentially less brittle to iterate over `examples/**/Cargo.toml` and run `cargo add shuttle-service@$version`?

File: publish.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    echo $version | rg "\d+\.\d+\.\d+" || { echo "first argument must be in the form x.y.z"; exit 1; }
+
+    publish-folder "common"
+    publish-folder "codegen"
+    publish-folder "service"
+    publish-folder "cargo-shuttle"
+
+    git tag "v$version"
+    git push --tags
```
Comment (brokad) [inline publish.sh:32]: May want to just `git push v$version` instead of all tags.

**Comment by brokad** [comment]
Can we also maybe move this to a `scripts/`?

File: scripts/publish.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+#
+# Little script to publish to crates.io
+# Usage: publish.sh x.y.z
+#
+# Dependencies: git, ripgrep
```
Comment (chesedo) [inline scripts/publish.sh:6]: I'm a bit lost where `cargo set-version` is used in this script :sweat_smile:

Comment (brokad) [reply scripts/publish.sh:6]: Haha oops, sorry it's only used in the other one! :smile:

**Review by brokad — APPROVED** [review]

**Pull request merged**